core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -56,7 +57,10 @@ public static boolean hasDeletes(FileScanTask task) {
     Preconditions.checkArgument(lookback > 0, "Invalid split planning lookback (negative or 0): %s", lookback);
     Preconditions.checkArgument(openFileCost >= 0, "Invalid file open cost (negative): %s", openFileCost);
 
-    Function<FileScanTask, Long> weightFunc = file -> Math.max(file.length(), openFileCost);
+    // For V2Format, we should check the size of delete file as well to avoid unbalanced bin-packing
```

Comment:
No need to use personal pronouns. This should be simply "check the size ... to avoid ..."